src/lighteval/metrics/normalizations.py

```diff
@@ -352,6 +358,60 @@ def gsm8k_normalizer(text: str) -> str:
         return INVALID_ANS
 
 
+PUNCT = {chr(i) for i in range(sys.maxunicode) if unicodedata.category(chr(i)).startswith("P")}.union(
+    string.punctuation
+)
+
+_ARTICLE_PATTERNS = {
+    Language.english: r"\b(a|an|the)\b",
+    Language.spanish: r"\b(el|la|los|las|un|una|unos|unas)\b",
+    Language.portuguese: r"\b(o|a|os|as|um|uma|uns|umas)\b",
+    Language.italian: r"\b(il|lo|la|i|gli|le|un|uno|una)\b",
+    Language.french: r"\b(le|la|les|l'|un|une|des)\b",
+    Language.german: r"\b(der|die|das|den|dem|des|ein|eine|einer|eines|einem|einen)\b",
+    Language.finnish: r"\b(se|yksi|yks)\b",
+    Language.greek: r"\b(ὁ|οἱ|τοῦ|τῶν|τόν|τούς|ὦ|ἡ|αἱ|τῆς|τῶν|τήν|τάς|τό|τά|τοῦ|τῶν|τό|τά)\b",
+    Language.norwegian: r"\b(en|ei|et|den|det|de)\b",
+    Language.swedish: r"\b(en|ett|den|det|de)\b",
+    Language.turkish: r"\b(bir)\b",
+    Language.dutch: r"\b(de|het|een)\b",
+    Language.hungarian: r"\b(a|az|egy)\b",
+    Language.catalan: r"\b(el|la|els|les|un|una|uns|unes)\b",
+    Language.hebrew: r"\b(ה)\b",
+    Language.galician: r"\b(o|a|os|as|un|unha|uns|unhas)\b",
+}
+
+
+def remove_articles(text: str, lang: Language) -> str:
+    """
+    Removes definite and indefinite articles from the text.
+    Generated using LLM then manually checked by non-expert.
+    Only languages that don't blend the articles, if you are native speaker,
+    we would appreciate adding also languages that blend the articles.
+    """
+    pattern = _ARTICLE_PATTERNS.get(lang)
+    return re.sub(pattern, " ", text) if pattern else text
+
+
+def remove_punc(text: str) -> str:
+    return "".join(ch for ch in text if ch not in PUNCT)
+
+
+def get_multilingual_normalizer(lang: Language, lower: bool = True) -> Callable[[str], str]:
+    tokenizer = get_word_tokenizer(lang)
+
+    def _inner_normalizer(text: str) -> str:
+        text = remove_articles(text, lang)
+        text = remove_punc(text)
+        if lower:
+            text = text.lower()
+
+        tokens = tokenizer.word_tokenize(text)
```

Comment:
So do we assume the user will pass the same custom tokenizer when using their model? Is there a check for this somewhere?

Reply:
No I think the tokenizer is choosen from the lang only, user cannot use their own for the metrics

Reply:
Rewrote my point above for clarity. I agree with you and think that it might cause issues if models are using a tokenizer and metrics are using another tokenizer, and that there is no way to check for match or mismatch of the tokenized sentences between.

Reply:
The purpose of tokenizers is completely different:
- Models use tokenizers for conversion from text to numbers
- Metrics use tokenizers for word separation (for computing f1)

I don't see any problem here there is not mismatch as the tokenizers are compeltely different.

Reply:
Don't we expect the model tokenizer and evaluation tokenizer to behave similarly though?